hyp3_rtc_gamma/rtc_sentinel.py

```diff
@@ -38,15 +39,28 @@
 from hyp3_rtc_gamma.xml2meta import sentinel2meta
 
 
-def get_product_name(granule_name, resolution=30, gamma0=True, power=True, filtered=False):
+def get_product_name(granule_name, orbit_file=None, resolution=30, power=True, filtered=False, gamma0=True):
     platform = granule_name[0:3]
     beam_mode = granule_name[4:6]
+    polarization = granule_name[14:16]
     datetime = granule_name[17:32]
-    g = 'g' if gamma0 else 's'
+
+    if orbit_file is None:
+        o = 'O'
+    elif 'POEORB' in orbit_file:
+        o = 'P'
+    elif 'RESORB' in orbit_file:
+        o = 'R'
+    else:
+        o = 'O'
```

Comment:
Open to suggestions here, not in love with this implementation.

Reply:
I agree, but nothing much better comes to mind
```python
replacements = {
    'POEORB': 'P',
    'RESORB': 'R',
    'default': 'O'
}
if orbit_file is not None:
    o = [v for k, v in replacements if k in orbit_file][0]
if o is not None:
    o = 'O'
```
for example is much worse